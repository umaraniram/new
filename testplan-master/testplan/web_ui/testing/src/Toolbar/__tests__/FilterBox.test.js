import React from 'react';
import { shallow } from 'enzyme';
import { StyleSheetTestUtils } from "aphrodite";

import FilterBox from '../FilterBox';

function defaultProps() {
  return {
    width: '19.5em',
    handleNavFilter: jest.fn(),
  };
}

describe('FilterBox', () => {
  let props;
  let mountedFilterBox;
  const renderFilterBox = () => {
    if (!mountedFilterBox) {
      mountedFilterBox = shallow(
        <FilterBox {...props} />
      );
    }
    return mountedFilterBox;
  };

  beforeEach(() => {
    // Stop Aphrodite from injecting styles, this crashes the tests.
    StyleSheetTestUtils.suppressStyleInjection();
    props = defaultProps();
    mountedFilterBox = undefined;
  });

  afterEach(() => {
    // Resume style injection once test is finished.
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
    props.handleNavFilter.mockClear();
  });

  it('shallow renders without crashing', () => {
    renderFilterBox();
  });

  it('shallow renders the correct HTML structure', () => {
    const filterBox = renderFilterBox();
    expect(filterBox).toMatchSnapshot();
  });

  it('calls handleNavFilter when onKeyUp event is triggered ' +
     'for input box', () => {
    const handleNavFilter = props.handleNavFilter;
    const filterBox = renderFilterBox();

    filterBox.find('input').simulate('keyUp', {keyCode: 13, key: 'Enter'});
    expect(handleNavFilter.mock.calls.length).toEqual(1);
  })

});