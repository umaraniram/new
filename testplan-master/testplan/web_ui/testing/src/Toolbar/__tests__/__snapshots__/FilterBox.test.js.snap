// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterBox shallow renders the correct HTML structure 1`] = `
<div
  className="searchBox_1yfus1e"
  style={
    Object {
      "width": "19.5em",
    }
  }
>
  <input
    className="searchBoxInput_kcbf9x"
    onKeyUp={[MockFunction]}
    placeholder="ï€‚ Filter"
    type="text"
  />
</div>
`;
